### PR TITLE
RAB-1171: Handle null Pagination object in SqlFindJobInstances

### PR DESCRIPTION
**Pull request by axelhms**
**Description (for Contributor and Core Developer)**

In this PR, I've added the possibility to have a null Pagination object in SqlFindJobInstances.

**Definition Of Done (for Core Developer only)**

- [ ] Tests
- [ ] Migration & Installer
- [ ] PM Validation (Story)
- [ ] Changelog (maintenance bug fixes)
- [ ] Tech Doc

### CODE REVIEW

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php [hunk 1]
```diff
@@ -83,30 +83,49 @@ private function buildPaginationPart(JobInstanceQueryPagination $queryPagination
...
+            $queryTypes,
         )->fetchAllAssociative();
 
         return array_map(
             static fn (array $jobInstance) => new JobInstance($jobInstance['code'], $jobInstance['label']),
             $results,
         );
     }
+
+    private function buildQueryParametersAndTypes(JobInstanceQuery $query): array
```
Comment (axelhms) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php:102]: There's probably a cleaner way to do this function, feel free to propose ideas

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php [hunk 2]
```diff
@@ -44,7 +44,7 @@ private function buildSqlQuery(JobInstanceQuery $query): string
 SQL;
 
         $sqlWherePart = $this->buildWherePart($jobNames, $search);
-        $sqlPaginationPart = $this->buildPaginationPart($pagination);
+        $sqlPaginationPart = null !== $pagination ? $this->buildPaginationPart($pagination) : '';
```
Comment (ValentinMumble) [inline src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php:47]: I'd put the null check in the `buildPaginationPart` function, but not really important

**Review by ValentinMumble — APPROVED** [review]

File: src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php (hunk 2 above)
Comment (axelhms) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php:47]: We could but we'd have to make the `buildPaginationPart` parameter nullable, not sure if that's worth it ?

Comment (ghoussard) [reply src/Akeneo/Platform/Job/back/Infrastructure/Query/SqlFindJobInstance.php:47]: Not a problem for me : as `buildPaginationPart` take a `JobInstanceQueryPagination` which is already nullable in the query

**Pull request merged**